### ISSUE TITLE
Import Error: Fontconfig not found

### ISSUE THREAD

**Issue by tnekent** [vispy/vispy#1556]
If I try to import vispy, it raises ImportError "fontconfig not found". Is there something missing?

**Comment by kmuehlbauer** [comment]
How did you install VisPy? Seems, that `fontconfig` is not installed.

**Comment by tnekent** [comment]
Installed through terminal with "pip install vispy"

**Comment by kmuehlbauer** [comment]
OK. Can you check with your system's package manager if `fontconfig` is installed in your system? If not, check if installation of `fontconfig` will fix it.

**Comment by tnekent** [comment]
How do you install fontconfig?

**Comment by kmuehlbauer** [comment]
What system are you on? You would use your system package manager (eg. apt, zypper etc. depending on system) and install `fonctconfig` through this.

**Comment by kmuehlbauer** [comment]
@djhoese @larsoner This seems to be a problem for users. `fontconfig` isn't mentioned in the docs, if it is missing, users will get these notifications. But we also have this patch currently in `conda-forge` to get things running (https://github.com/conda-forge/vispy-feedstock/blob/master/recipe/ctypes_fontconfig.diff). 

Can we do anything to correctly find `fontconfig` for the different systems? See https://github.com/vispy/vispy/blob/master/vispy/ext/fontconfig.py

**Comment by larsoner** [comment]
I think that code was originally taken from glumpy, maybe there are hints over there

**Comment by kmuehlbauer** [comment]
@larsoner Thanks for looking into this. But glumpy doesn't use `fontconfig`.

**Comment by djhoese** [comment]
@kmuehlbauer If I'm not mistaken that conda-forge patch is to make sure it takes the fontconfig from conda and not from the system. It isn't necessarily a fix for anything wrong in vispy.

I *think* the TextVisuals require fontconfig to be installed to find and load font glyphs. We could make it not raise an ImportError until a TextVisual is actually made rather than when it is imported (which is always because of how the visuals are imported).

**Comment by kmuehlbauer** [comment]
@djhoese Wasn't meant to blame vispy doing something wrong. But currently we do not know, if `fontconfig` is really missing in the OP's system or if vispy just isn't finding it.

**Comment by kmuehlbauer** [comment]
Does this have a solution yet, @djhoese ?

**Comment by djhoese** [comment]
I don't think so. I believe the conda-forge package depends on fontconfig so this becomes a non-issue for that particular installation method. I'm not sure there is much we can do on the PyPI pip side of things, right? We now build wheels, but since we are dynamically loading the library that shouldn't matter (right?). I'm not sure of a better way to depend on this C library without making our own python library that...installs it in to the python environment?

Am I missing anything?

I still say we could provide a better error message that only appears when the TextVisual is made and I'm not sure we are doing that yet.

**Comment by kmuehlbauer** [comment]
It looks like we did not take actions so far, so I'll have a look into this next week.

**Comment by kmuehlbauer** [comment]
It seems I can't reproduce this on my linux system, where fontconfig is installed per default.

**Comment by djhoese** [comment]
I think the error only occurs on systems that don't have fontconfig installed and the user didn't install from conda-forge. The error says fontconfig not found. Should we update it to say something like "The C library fontconfig is required to use TextVisuals, please use your system package manager or other installation method to install it"?

**Comment by darkvertex** [comment]
Experienced this while using the Debian based `python:3.10-slim` Docker image in a StableDifussion deployment.

Had to `apt-get install -y fontconfig` and then it said:
```
WARNING: Could not load OpenGL library.
```
so I did `apt-get install -y libglfw3-dev libgles2-mesa-dev` and it finally worked. 🤷‍♂️ 


Someone ought to document these mystery dependencies in the README.

**Comment by djhoese** [comment]
Maybe not the README, but the installation instructions:

https://vispy.org/installation.html

Unfortunately these GL dependencies all fall under the umbrella of "install OpenGL on your system" and are different for each system and even for each version of a distribution (ex. Ubuntu version).

But yeah the fontconfig dependency (which is only needed for text right?) should definitely be added. Pull requests are welcome to add this information to the installation instructions.
